test/functional/legacy/082_spec.lua

```diff
@@ -0,0 +1,105 @@
+-- Tests for case-insensitive UTF-8 comparisons (utf_strnicmp() in mbyte.c)
+-- Also test "g~ap".
+
+local helpers = require('test.functional.helpers')
+local feed, insert, source = helpers.feed, helpers.insert, helpers.source
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('82', function()
+  setup(clear)
+
+  it('is working', function()
+    execute('set enc=utf8')
+    feed('ggdG<cr>')
+    execute('function! Ch(a, op, b, expected)')
```

Comment:
You can place helper functions in a `source(...)` block to make the test easier to read. See other tests for examples.

For tests like this one that don't actually exercise the UI functionality (buffer, cursor, ...) you could actually put almost everything inside `source`. See eg `095_regexp_multibyte_spec.lua`.